ansible_events_ui/api/activation.py

```diff
@@ -1,8 +1,125 @@
 """Activation API endpoints."""
-from fastapi import APIRouter
+import logging
+import json
+
+from fastapi import APIRouter, Depends
+import asyncio
+
+import sqlalchemy.orm
+from sqlalchemy import insert, select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ansible_events_ui.db.dependency import (
+    get_db_session,
+    get_db_session_factory,
+)
+from ansible_events_ui.db.models import (
+    activations,
+    extra_vars,
+    inventories,
+    rule_set_files,
+    activation_instance_logs,
+    playbooks,
+    execution_envs,
+)
+from ansible_events_ui.managers import (
+    taskmanager,
+    connectionmanager,
+)
+from ansible_events_ui.ruleset import (
+    activate_rulesets,
+)
+from ansible_events_ui.schemas import (
+    Activation,
+)
+
+logger = logging.getLogger("ansible_events_ui")
 
 __all__ = ("router",)
 
 router = APIRouter()
 
-# Implement activation API here.
+@router.post("/api/activation/")
+async def create_activation(
+    a: Activation,
+    db: AsyncSession = Depends(get_db_session),
+    db_session_factory: sqlalchemy.orm.sessionmaker = Depends(
+        get_db_session_factory
+    ),
+):
+    query = select(rule_set_files).where(
+        rule_set_files.c.id == a.rule_set_file_id
+    )
+    rule_set_file_row = (await db.execute(query)).first()
```

Comment:
It's good place to start doing error handling appropriately.
This endpoint should return status `422 Unprocessable Entity` when referenced resource doesn't exist (`rule_set_file`, `inventory`, `extra_var`, `playbook` or `execution_env`).